source/include/stdbool.readme

```diff
@@ -0,0 +1,30 @@
+#ifndef _STDBOOL_H
+#define _STDBOOL_H
+
+/*******************************************************************************
+ * This file contains the definitions specified in stdbool.h. It is provided to
+ * allow the library to be built using compilers that do not provide their own
+ * stdbool.h defintion.
+ *
+ * To use this file:
+ *
+ *    1) Copy this file into a directory that is in your compiler's include path.
+ *       The directory must be part of the include path for system header files,
+ *       for example passed using gcc's "-I" or "-isystem" options.
+ *
+ *    2) Rename the copied file stdbool.h.
+ *
+ */
+
+#ifndef __cplusplus
```

Comment:
Can we use [the standard way to test for C99 compatibility in C code](https://en.wikipedia.org/wiki/C99#Version_detection) here?

```c
#if __STDC_VERSION__ < 199901L
/* _Bool was introduced in C99. */
    #define bool     int
    #define false    0
    #define true     1
#endif

```

Reply:
We had something like that in coreMQTT, but Richard spoke out against it (you can see the thread in Slack).  The solution, as implemented in coreMQTT, was the outcome of that discussion.  This PR seeks symmetry with the coreMQTT solution.